src/core/thread/topology.cpp

```diff
@@ -261,6 +261,44 @@ bool Child::HasIp6Address(const Ip6::Address &aAddress) const
     return retval;
 }
 
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+void Child::RemovePreviousDomainUnicastAddress(void)
+{
+    for (const Ip6::Address &ip6Address : mIp6Address)
+    {
+        VerifyOrExit(!ip6Address.IsUnspecified(), OT_NOOP);
+
+        if (Get<BackboneRouter::Leader>().IsPreviousDomainUnicast(ip6Address))
+        {
+            IgnoreError(RemoveIp6Address(ip6Address));
+            ExitNow();
+        }
+    }
+
+exit:
+    return;
+}
+
+otError Child::GetDomainUnicastAddress(Ip6::Address &aAddress) const
+{
+    otError error = OT_ERROR_NOT_FOUND;
+
+    for (const Ip6::Address &ip6Address : mIp6Address)
+    {
+        VerifyOrExit(!ip6Address.IsUnspecified(), OT_NOOP);
+
+        if (Get<BackboneRouter::Leader>().IsDomainUnicast(ip6Address))
+        {
+            aAddress = ip6Address;
```

Comment:
We have one copy here. 
Is it possible that we avoid this copy?
For example, by changing this method to:
```c++
const Ip6::Address *Child::GetDomainUnicastAddress(void) const
```

Reply:
If returning a pointer, the content will be cleared before process AR TLV in  `MleRouter::UpdateChildAddresses()`, thus no way to compare the old one and new one.

Reply:
After getting the pointer, the caller can decide to copy the address or not according to its intention?

Reply:
no strong preference, I am fine to change it if you like it.